package.json

```diff
@@ -26,14 +26,15 @@
     "snarkjs": "^0.6.11",
     "solhint-plugin-prettier": "^0.0.5",
     "solidity-coverage": "^0.8.2",
-    "solidity-rlp": "^2.0.5"
+    "solidity-rlp": "^2.0.5",
+    "solc": "^0.8.25"
   },
   "scripts": {
     "compile": "hardhat compile",
     "test": "hardhat test",
     "prettier:check": "prettier-check contracts/**/*.sol",
     "prettier:fix": "prettier --write contracts/**/*.sol test/**/*.js scripts/**/*.js",
-    "deploy": "npx hardhat run --network sepolia scripts/deploy.js"
+    "deploy": "npx hardhat times && npx hardhat run --network sepolia scripts/deploy.js && npx hardhat undo"
```

Comment:
Is the hardhat able to verify the source code after we undo the startTime in the source code?

Reply:
The startTime in the .sol file is now a flag and cannot be changed manually. When the contract is deployed, the flag will be automatically replaced with the current time, and the modification will be revoked after deployment.

Reply:
There is no problem in deploying the script after manually modifying the startTime. Because the flag cannot be found, the specific logic of replacement and revocation is not executed.
However, the test script will report an error after modification.

Reply:
```
if (data.indexOf(startTime) === -1) {
    return;
 }
```

Reply:
The code can be automatically verified when deploying the contract.